app/pages/project/project-page.spec.js

```diff
@@ -130,4 +132,36 @@ describe('ProjectPage', () => {
       expect(fieldguide).to.have.lengthOf(1);
     });
   });
+
+  describe('on component lifecycle', () => {
+    const sugarClientSubscribeSpy = sinon.spy(sugarClient, 'subscribeTo');
+    const sugarClientUnsubscribeSpy = sinon.spy(sugarClient, 'unsubscribeFrom');
+    const channel = `project-${project.id}`;
+    let wrapper;
+    beforeEach(() => {
+      wrapper = mount(
+        <ProjectPage project={project}>
+          <Page />
+        </ProjectPage>
+      );
+    });
+
+    it('subscribes the user to the sugar project channel on mount', () => {
+      expect(sugarClientSubscribeSpy.calledWith(channel)).to.equal(true);
+    });
+
+    it('unsubscribes the user from the sugar project channel on unmount', () => {
+      wrapper.unmount();
+      expect(sugarClientUnsubscribeSpy.calledWith(channel)).to.equal(true);
+    });
+
+    it('unsubscribes old project and subscribes new project on project change', () => {
+      const newProject = {id: '999', title: 'fake project', slug: 'owner/name'};
+      const newChannel = `project-${newProject.id}`;
+      sugarClientSubscribeSpy.reset()
```

Comment:
You should reset your unsubscribe Spy here too.

Reply:
is [profile.spec](https://github.com/zooniverse/Panoptes-Front-End/blob/3c689a89e4e45d1f166afbd24c20ace7cbe0e8cc/app/pages/settings/profile.spec.js) a good reference?

Reply:
Maybe. I wasn't familiar with sinon when I wrote those. I'd read more of the sinon docs when I wrote the workflow selection tests for the classifier.

Reply:
https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/project/index.spec.js